quinn-udp/src/unix.rs

```diff
@@ -325,12 +358,64 @@ fn send(
     }
 }
 
-#[cfg(any(
-    target_os = "macos",
-    target_os = "ios",
-    target_os = "openbsd",
-    target_os = "netbsd"
-))]
+#[cfg(any(target_os = "macos", target_os = "ios",))]
+fn send(state: &UdpSocketState, io: SockRef<'_>, transmit: &Transmit<'_>) -> io::Result<()> {
+    let mut hdrs = unsafe { mem::zeroed::<[msghdr_x; BATCH_SIZE]>() };
+    let mut iovs = unsafe { mem::zeroed::<[libc::iovec; BATCH_SIZE]>() };
+    let mut ctrls = [cmsg::Aligned([0u8; CMSG_LEN]); BATCH_SIZE];
+    let addr = socket2::SockAddr::from(transmit.destination);
+    let segment_size = transmit.segment_size.unwrap_or(transmit.contents.len());
+    let cnt = transmit
+        .contents
+        .chunks(segment_size)
+        .enumerate()
+        .map(|(i, chunk)| {
+            prepare_msg(
+                &Transmit {
+                    destination: transmit.destination,
+                    ecn: transmit.ecn,
+                    contents: chunk,
+                    segment_size: Some(chunk.len()),
+                    src_ip: transmit.src_ip,
+                },
+                &addr,
+                &mut hdrs[i],
+                &mut iovs[i],
+                &mut ctrls[i],
+                true,
+                state.sendmsg_einval(),
+            );
+            hdrs[i].msg_datalen = chunk.len();
+        })
+        .count();
+    let n = unsafe { sendmsg_x(io.as_raw_fd(), hdrs.as_ptr(), cnt as u32, 0) };
+    // print!("o{} ", n);
+    if n == -1 {
```

Comment:
Nit: please flip the condition and make this an early return.

Reply:
This always returns `Ok(())`, just as in the BSD version below. (Should both of these actually return an Error in this case?)

Reply:
> Nit: please flip the condition and make this an early return.

Agreed with @larseggert. I suggest keeping as is, as both paths (`== -1` and `>-1`) have the same return value `Ok(())` and given that it mirrors the BSD implementation below.

> Should both of these actually return an Error in this case?

In case we do want to return an error here, I suggest doing this as part of https://github.com/quinn-rs/quinn/issues/1971 across all platforms (e.g. as you mention BSD) and not in this pull request.

Reply:
> Agreed with @larseggert. I suggest keeping as is, as both paths (`== -1` and `>-1`) have the same return value `Ok(())` and given that it mirrors the BSD implementation below.

I don't find this a very strong argument. This saves some rightward drift for the error handling code and duplicating the `Ok(())` return value doesn't seem like a particularly big deal here. @Ralith opinions?